scripts/testing/programs/match_template/test_template_reproducibility.py

```diff
@@ -26,12 +26,46 @@
 import os
 import shutil
 import sys
+import re
 
 # By default the "_gpu" suffix will be added unless the --old-cistem flag is used
 # or the --cpu flag is used
 wanted_binary_name = 'match_template'
 
 
+def extract_threshold_value(hist_file_path):
+    """
+    Extract the threshold value from the histogram text file.
+    
+    The threshold value is in the first line of the file, which starts with
+    "# Expected". This function extracts the numerical value following this marker.
+    
+    Args:
+        hist_file_path (str): Path to the histogram text file
+        
+    Returns:
+        float: The extracted threshold value
+        
+    Raises:
+        FileNotFoundError: If the file doesn't exist
+        ValueError: If the threshold value cannot be found or parsed
+    """
+    if not exists(hist_file_path):
+        raise FileNotFoundError(f"Histogram file not found at {hist_file_path}")
+    
+    with open(hist_file_path, 'r') as f:
+        first_line = f.readline().strip()
+        
+    # Use regex to extract the threshold value from the line
+    # Looking for a pattern like "# Expected max is: 0.12345"
+    match = re.search(r'# Expected max is: ([\d.e+-]+)', first_line)
```

Comment:
I think this line is breaking things. The pattern is actually like this code block

`# Expected threshold = 6.90`

So we just need the value after the equals sign. Normally I would do this in something simple like awk, but I'm not sure what is best in python. I've also used numpy get text in python before.